steering_controllers_library/src/steering_odometry.cpp

```diff
@@ -73,70 +73,47 @@ bool SteeringOdometry::update_odometry(
 bool SteeringOdometry::update_from_position(
   const double traction_wheel_pos, const double steer_pos, const double dt)
 {
-  /// Get current wheel joint positions:
-  const double traction_wheel_cur_pos = traction_wheel_pos * wheel_radius_;
-  const double traction_wheel_est_pos_diff = traction_wheel_cur_pos - traction_wheel_old_pos_;
+  const double traction_wheel_est_pos_diff = traction_wheel_pos - traction_wheel_old_pos_;
 
   /// Update old position with current:
-  traction_wheel_old_pos_ = traction_wheel_cur_pos;
+  traction_wheel_old_pos_ = traction_wheel_pos;
 
-  /// Compute linear and angular diff:
-  const double linear_velocity = traction_wheel_est_pos_diff / dt;
-  steer_pos_ = steer_pos;
-  const double angular_velocity = tan(steer_pos) * linear_velocity / wheelbase_;
-
-  return update_odometry(linear_velocity, angular_velocity, dt);
+  return update_from_velocity(traction_wheel_est_pos_diff / dt, steer_pos, dt);
```

Comment:
In the update from velocity, why are we passing `steer_pos` instead of it's velocity (angular)?

Reply:
Here?

Reply:
"from_velocity" refers to the encoder on the traction wheels. steering is always from position
https://github.com/ros-controls/ros2_controllers/blob/d8ce400c2b5d9716c63f2563df212a3432046a69/steering_controllers_library/include/steering_controllers_library/steering_odometry.hpp#L96-L104